### PR TITLE
neutron contract rpc

### PR DESCRIPTION
**Pull request by icodeface**

### CODE REVIEW

**Comment by icodeface** [comment]
still some issue("bad-tx-improper-version-0" error in validation.cpp) , it needs to be fixed in https://github.com/qtumproject/qtum-neutron/pull/4

**Review by xuanyan0x7c7 — APPROVED** [review]

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1281,6 +1283,101 @@ UniValue callcontract(const JSONRPCRequest& request)
...
+        throw std::runtime_error(
+                RPCHelpMan{"neutroncallcontract",
+                           "\nCall contract methods offline.\n",
+                           {
+                                   {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The contract address"},
+                                   {"data", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The data hex string"},
+                                   {"senderAddress", RPCArg::Type::STR, RPCArg::Optional::OMITTED_NAMED_ARG, "The sender address string"},
+                                   {"gasLimit", RPCArg::Type::NUM, RPCArg::Optional::OMITTED_NAMED_ARG, "The gas limit for executing the contract."},
+                           },
+                           RPCResult{
```
Comment (Earlz) [inline src/rpc/blockchain.cpp:1298]: This is ok for now, but will need to be a different format later: https://gist.github.com/Earlz/aa235c9fa9258ae6035c26ebcd90cfe6#neutronexecutionresult

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -1281,6 +1283,101 @@ UniValue callcontract(const JSONRPCRequest& request)
...
+//    if(!globalState->addressInUse(addrAccount))
+//        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Address does not exist");
+
+    std::string data = request.params[1].get_str();
+    if(data.size() % 2 != 0 || !CheckHex(data))
+        throw JSONRPCError(RPC_TYPE_ERROR, "Invalid data (data not hex)");
+
+    UniversalAddress senderAddress;
+    if(request.params.size() >= 3){
+        if (!ReadUniversalAddress(request.params[2].get_str(), senderAddress) || !IsValidUniversalAddress(senderAddress)) {
```
Comment (Earlz) [inline src/rpc/blockchain.cpp:1355]: senderAddress should accept both UniversalAddress and standard (pubkeyhash or scripthash) base58 addresses

File: src/validation.cpp [hunk 3]
```diff
@@ -801,9 +801,9 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
                     txMinGasPrice = qtumTransaction.gasPrice();
                 }
                 VersionVM v = qtumTransaction.getVersion();
-                if(v.format!=0)
+                if (v.format > 1 || (v.format == 1 && chainActive.Tip()->nHeight < Params().GetConsensus().NeutronHeight))
                     return state.DoS(100, error("AcceptToMempool(): Contract execution uses unknown version format"), REJECT_INVALID, "bad-tx-version-format");
-                if(v.rootVM != 1)
+                if(!((v.format == 0 && v.rootVM == 1) || (v.format == 1 && (v.rootVM == 3 || v.rootVM == 4))))
```
Comment (Earlz) [inline src/validation.cpp]: The 1/3/4 numbers should use constants so that their meaning is clear ie `...v.rootVM == NEUTRON_ROOTVM_X86` or similar

**Review by Earlz — CHANGES_REQUESTED** [review]
Just some minor changes needed

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (icodeface) [reply src/rpc/blockchain.cpp:1355]: `ReadUniversalAddress()` can accept base58/bech32/universal hex address

**Review by Earlz — APPROVED** [review]

**Comment by icodeface** [comment]
@Earlz resolved the merge conflicts, can u review it

**Pull request merged**